### PR TITLE
Add tests for timeoutPromise

### PR DESCRIPTION
**Pull request by JonathanDCohen**
Also specified the typing of the Promises instead of using Promise<any>

### CODE REVIEW

File: src/www/app/util.spec.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+describe('timeoutPromise', () => {
+  it('Executes successful promise', () => {
+    timeoutPromise(Promise.resolve(PROMISE_RESOLVED), 100, 'Test Promise')
+        .catch((err) => {
+          fail(`Successful promise was timed out when it should have resolved`);
+        })
+        .then((value) => {
+          expect(value).toEqual(PROMISE_RESOLVED);
+        });
+    const promiseWithTime = new Promise((resolve, _) => {
```
Comment (alalamav) [inline src/www/app/util.spec.ts]: This could be a separate test, something like "Executes promise within timeout".

File: src/www/app/util.ts [hunk 2]
```diff
@@ -14,13 +14,15 @@
...
 import * as errors from '../model/errors';
 
-// tslint:disable-next-line:no-any
-export function timeoutPromise(promise: Promise<any>, ms: number, name = '') {
-  // tslint:disable-next-line:no-any
-  let winner: Promise<any>;
+// Attempts to settle `promise`, but times it out after `ms` milliseconds.  A failed return type
+// could either be a timed-out Promise or a failed Promise.  Use an instanceof guard to
+// differentiate between the two.
+export function timeoutPromise<T>(promise: Promise<T>, ms: number, name = ''): Promise<void|T> {
```
Comment (alalamav) [inline src/www/app/util.ts]: Very nice how you got rid of `any` 👍

File: src/www/app/util.spec.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  });
+
+  it('Executes failed promise', () => {
+    timeoutPromise(Promise.reject('reason'), 100, 'Test Promise').catch((err) => {
+      expect(err).not.toEqual(jasmine.any(OperationTimedOut));
+    });
+    const promiseWithTime = new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(PROMISE_RESOLVED);
+      }, 50);
```
Comment (alalamav) [inline src/www/app/util.spec.ts]: nit: maybe use a constant for the timeout values, with multipliers when needed.

File: src/www/app/util.spec.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    });
+    timeoutPromise(promiseWithTime, 100, 'Test Promise').catch((err) => {
+      expect(err).not.toEqual(jasmine.any(OperationTimedOut));
+    });
+  });
+
+  it('Times out promise', () => {
+    const promiseWithTime = new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(1);
```
Comment (alalamav) [inline src/www/app/util.spec.ts]: nit: `PROMISE_RESOLVED`

File: src/www/app/util.spec.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  it('Executes failed promise', () => {
+    timeoutPromise(Promise.reject('reason'), 100, 'Test Promise').catch((err) => {
+      expect(err).not.toEqual(jasmine.any(OperationTimedOut));
+    });
+    const promiseWithTime = new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(PROMISE_RESOLVED);
+      }, 50);
+    });
+    timeoutPromise(promiseWithTime, 100, 'Test Promise').catch((err) => {
```
Comment (alalamav) [inline src/www/app/util.spec.ts]: Why do you expect the promise to fail? The promise should have been resolved, since it's under the timeout.

Comment (JonathanDCohen) [reply src/www/app/util.spec.ts]: Because that's supposed to be a failing promise

**Pull request closed without merging**